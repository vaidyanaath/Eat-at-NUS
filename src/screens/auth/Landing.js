import React from 'react';
import { StatusBar, StyleSheet, Image, View } from 'react-native';

// import components
import { InnerContainer } from '../../components/containers/InnerContainer';
import { ColoredButton } from '../../components/buttons/ColoredButton';
import { BigText } from '../../components/texts/BigText';
import { RegularText } from '../../components/texts/RegularText';

// import colors
import { colors } from '../../assets/colors';

const Landing = ({ navigation }) => {
  const handleIsCustomer = () => {
    navigation.navigate('CustomerSignIn');
  };
  const handleIsFoodStallOwner = () => {};

  return (
    <InnerContainer style={styles.container}>
      <StatusBar barStyle="dark-content" backgroundColor={colors.bg} />

      <Image source={require('../../assets/images/logo_sidebar.png')} style={styles.logo} />
      <BigText style={styles.welcomeText}> Eat@NUS </BigText>

      <RegularText style={styles.subText}> I am a . . . </RegularText>
      <ColoredButton onPress={handleIsCustomer} style={styles.customerButton}>
        <BigText style={styles.buttonText}> Customer </BigText>
      </ColoredButton>
      <ColoredButton onPress={handleIsFoodStallOwner} style={styles.foodStallOwnerButton}>
        <BigText style={styles.buttonText}> Food Stall Owner </BigText>
      </ColoredButton>
    </InnerContainer>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'column',
    padding: 0,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: colors.bg,
  },
  welcomeText: {
    fontSize: 35,
    fontFamily: 'SourceSansPro-Bold',
    marginTop: 10,
    marginBottom: 60,
    textAlign: 'center',
    color: colors.white,
    justifyContent: 'flex-end',
  },
  logo: {
    width: 300,
    height: 150,
    // backgroundColor: '#324',
  },
  subText: {
    fontSize: 30,
    marginBottom: 10,
    color: colors.secondary,
  },
  customerButton: {
    width: 280,
    height: 70,
    marginBottom: 10,
    backgroundColor: colors.secondary,
  },
  foodStallOwnerButton: {
    width: 280,
    height: 70,
    marginBottom: 10,
    backgroundColor: colors.primary,
  },
  buttonText: {
    color: colors.gray,
    fontSize: 25,
  },
});

export default Landing;
