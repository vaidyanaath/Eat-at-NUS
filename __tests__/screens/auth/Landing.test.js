import React from 'react';
import renderer from 'react-test-renderer';

import Landing from '../../../src/screens/auth/Landing';

describe('<Landing />', () => {
    
    it('Renders correctly', () => {
        const mockNavigation = { navigate: () => {} };
        const tree = renderer.create(<Landing navigation={mockNavigation}/>).toJSON();
        expect(tree).toMatchSnapshot();
    });
    
});