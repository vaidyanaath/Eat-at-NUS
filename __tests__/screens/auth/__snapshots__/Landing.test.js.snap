// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Landing /> Renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
        "width": "100%",
      },
      Object {
        "alignItems": "center",
        "backgroundColor": "#2C365A",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
        "padding": 0,
      },
    ]
  }
>
  <Image
    source={1}
    style={
      Object {
        "height": 150,
        "width": 150,
      }
    }
  />
  <Text
    style={
      Array [
        Object {
          "color": "#2C365A",
          "fontFamily": "SourceSansPro-SemiBold",
          "fontSize": 33,
        },
        Object {
          "color": "#FFFFFF",
          "fontFamily": "SourceSansPro-Bold",
          "fontSize": 35,
          "justifyContent": "flex-end",
          "marginBottom": 60,
          "marginTop": 10,
          "textAlign": "center",
        },
      ]
    }
  >
     Eat@NUS 
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#2C365A",
          "fontFamily": "SourceSansPro-Regular",
          "fontSize": 23,
        },
        Object {
          "color": "#FFFFFF",
          "fontSize": 30,
          "marginBottom": 10,
        },
      ]
    }
  >
     I am a . . . 
  </Text>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "#ff7f50",
        "borderRadius": 10,
        "elevation": 3,
        "height": 70,
        "justifyContent": "center",
        "marginBottom": 10,
        "marginLeft": 5,
        "marginRight": 5,
        "marginTop": 5,
        "opacity": 1,
        "paddingBottom": 10,
        "paddingLeft": 10,
        "paddingRight": 10,
        "paddingTop": 10,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.22,
        "shadowRadius": 2.22,
        "width": 280,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#2C365A",
            "fontFamily": "SourceSansPro-SemiBold",
            "fontSize": 33,
          },
          Object {
            "color": "#F5F5F5",
            "fontSize": 25,
          },
        ]
      }
    >
       Customer 
    </Text>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "#ff7f50",
        "borderRadius": 10,
        "elevation": 3,
        "height": 70,
        "justifyContent": "center",
        "marginBottom": 10,
        "marginLeft": 5,
        "marginRight": 5,
        "marginTop": 5,
        "opacity": 1,
        "paddingBottom": 10,
        "paddingLeft": 10,
        "paddingRight": 10,
        "paddingTop": 10,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.22,
        "shadowRadius": 2.22,
        "width": 280,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#2C365A",
            "fontFamily": "SourceSansPro-SemiBold",
            "fontSize": 33,
          },
          Object {
            "color": "#F5F5F5",
            "fontSize": 25,
          },
        ]
      }
    >
       Food Stall Owner 
    </Text>
  </View>
</View>
`;
